### PR TITLE
ci: add coverage job

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
The results are pushed to codecov.io and should be reported on PRs.

The only thing left is the integration subprocess tests seem to hang.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -596,8 +598,7 @@ basepython=python
 
 # DEV: We use `conftest.py` as a local pytest plugin to configure hooks for collection
 [pytest]
-# Common directories to ignore
-addopts = --ignore "tests/utils" --ignore "tests/base" --durations=10
+addopts = --cov=ddtrace/ --cov-append --cov-report= --durations=10
```
Comment (brettlangdon) [inline tox.ini:601]: do these get added to subprocess invocations as well?

**Comment by Kyle-Verhoog** [comment]
Coverage report (without integration tests): https://376242-61572326-gh.circle-artifacts.com/0/htmlcov/index.html

File: tox.ini (hunk 1 above)
Comment (Kyle-Verhoog) [reply tox.ini:601]: They should yes https://pytest-cov.readthedocs.io/en/latest/subprocess-support.html#subprocess-support

File: .circleci/config.yml [hunk 2]
```diff
@@ -82,6 +82,22 @@ commands:
...
 
+  save_codecov:
+    description: "Persist code coverage results"
+    steps:
+      # Make a coverage file if one wasn't generated
+      # persist_to_workspace will fail if no files are matched.
+      # Use `date` to come up with a unique name for the file
+      # to avoid collisions when workspaces are merged.
+      - run: touch .coverage
+      - run: mv .coverage ".coverage.report.$(date +%s%6N)"
```
Comment (brettlangdon) [inline .circleci/config.yml:93]: do we need to tell `coverage.py` and `codecov` to only look for this specific file/date?

Comment (brettlangdon) [reply .circleci/config.yml:93]: or no... this is the per-job report right?

Comment (Kyle-Verhoog) [reply .circleci/config.yml:93]: right, they get merged downstream in the coverage job into a single .coverage file which codecov should look for.

**Review by brettlangdon — APPROVED** [review]

**Pull request merged**